lib/browse_everything/driver/s3.rb

```diff
@@ -106,7 +113,13 @@ def bucket
       end
 
       def client
-        @client ||= Aws::S3::Client.new(credentials: Aws::Credentials.new(config[:app_key], config[:app_secret]), region: config[:region])
+        if @client.nil?
+          aws_config = {}
+          aws_config[:credentials] = Aws::Credentials.new(config[:app_key], config[:app_secret]) if config[:app_key].present?
+          aws_config[:region] = config[:region] if config.key?(:region)
```

Comment:
maybe an `aws_config` method instead that returns the hash and encapsulates the logic? Otherwise this looks good to me!